test/functional/tests/io/trim/test_trim_eviction.py

```diff
@@ -54,25 +56,49 @@ def test_trim_eviction(cache_mode, cache_line_size, filesystem, cleaning):
         core.create_filesystem(filesystem)
         core.mount(mount_point, ["discard"])
 
+    with TestRun.step("Create ioclass config."):
+        ioclass_config.create_ioclass_config()
+        ioclass_config.add_ioclass(
+            ioclass_id=1, eviction_priority=1, allocation="0.00", rule=f"metadata"
+        )
+        casadm.load_io_classes(
+            cache_id=cache.cache_id, file=ioclass_config.default_config_file_path
+        )
+
     with TestRun.step("Create random file using ddrescue."):
         test_file = fs_utils.create_random_test_file(test_file_path, core_dev.size * 0.9)
         create_file_with_ddrescue(core_dev, test_file)
+        os_utils.sync()
+        os_utils.drop_caches()
 
     with TestRun.step("Remove file and create a new one."):
         cache_iostats_before = cache_dev.get_io_stats()
+        data_reads_before = cache.get_io_class_statistics(io_class_id=0).block_stats.cache.reads
+        metadata_reads_before = cache.get_io_class_statistics(io_class_id=1).block_stats.cache.reads
         test_file.remove()
         os_utils.sync()
         os_utils.drop_caches()
         create_file_with_ddrescue(core_dev, test_file)
 
     with TestRun.step("Check using iostat that reads from cache did not occur."):
         cache_iostats_after = cache_dev.get_io_stats()
+        data_reads_after = cache.get_io_class_statistics(io_class_id=0).block_stats.cache.reads
+        metadata_reads_after = cache.get_io_class_statistics(io_class_id=1).block_stats.cache.reads
         reads_before = cache_iostats_before.sectors_read
         reads_after = cache_iostats_after.sectors_read
 
-        if reads_after != reads_before:
-            TestRun.fail(f"Number of reads from cache before and after removing test file "
-                         f"differs. Reads before: {reads_before}, reads after: {reads_after}.")
+        metadata_reads_diff = metadata_reads_after - metadata_reads_before
+        data_reads_diff = data_reads_after - data_reads_before
+        iostat_diff = (reads_after - reads_before) * cache_block_size
+
+        if iostat_diff > int(metadata_reads_diff) and int(data_reads_diff) > 0:
```

Comment:
I think that when this condition is met: ( int(data_reads_diff) > 0 ) test should fail automatically, not matter what is the metadata and iostat read difference